lib/wit/gitrepo.py

```diff
@@ -89,6 +95,17 @@ def clone(self, source, name):
             else:
                 raise
 
+    # Only newer git versions can use '--reference-if-able', so we emulate the 'if-able'.
+    def git_reference_capable(self):
+        env_ref_path = os.getenv("WIT_WORKSPACE_REFERENCE")
```

Comment:
I still really think we should do the environment handling in one place, rather than allowing any portion of the program grab at environment variables.

Also, you're missing a check to see whether this actually returned a None value or not, which means just below you are casting a `None` to a `Path`.